File: 482-License-Key-Formatting.py
class Solution:
    def licenseKeyFormatting(self, s: str, k: int) -> str:
        ms = s.replace("-","").upper()
        i = len(ms)%k or k
        j = 0
        ans = []
        while i <= len(ms):
            ans.append(ms[j:i])
            j = i
            i += k 
        return "-".join(ans)

File: 482-License-Key-Formatting.java
class Solution {
    public String licenseKeyFormatting(String s, int k) {
        String ms = s.replace("-","").toUpperCase();
        int i = ms.length() % k;
        if(i == 0) i = k;
        int j = 0;
        List<String> ans = new ArrayList<>();
        while(i<=ms.length()){
            ans.add(ms.substring(j,i));
            j = i;
            i += k;
        }
        return String.join("-",ans);
    }
}